### DISCUSSION TITLE
In Text web part, placement of migrated images not preserved through edit

### DISCUSSION THREAD

**Discussion by James6M** [WikiTransformationProject/wikitraccs-releases#89] [Q&A]
When a Confluence page has two images in a paragraph (and thus, appearing side-by-side), WikiTraccs nicely migrates them as floating images in a Text web part, one floating left and one floating right.  SharePoint displays them at the same vertical position, as they were in Confluence. (Although the second image now floats to the right, which may not have been true in the Confluence page.)

However, as soon as I try to edit the SharePoint page, this is immediately broken.  The second image moves vertically below the first image (as well as floating to the right side).  This arrangement appears in the editor, continues when the page is republished, and I can find no way to return except to roll back to the earlier version.  Is there any way to maintain the good WikiTraccs formatting through an edit of such a page?

**Comment by heinrich-ulbricht** [comment]
I need to check that. Microsoft is currently upgrading pages to a new format when editing pages. This already broke tables once.

Could you please send me the storage format of an affected page (if still available), plus screenshots before and after the edit? To contact @ wikitransformationproject.com

https://www.wikitransformationproject.com/docs/troubleshooting/troubleshooting-strategies/get-confluence-storage-format/

**Reply by heinrich-ulbricht** [reply]
Here's a video of how it should look when images are placed next to each other, and how editing the page should leave them in place:

https://github.com/WikiTransformationProject/wikitraccs-releases/assets/3469970/b2c46954-03ae-43ad-92b7-431ff5e85f8f

The video also shows how the developer tools are used to check the styles of one of the images. @James6M Could you please also do that (Right click an image -> Choose **Inspect**), and make a screenshot of what is shown, and send it to me? I hope to see whether the page content has been transformed to the new editor version or not. The new editor is not generally available yet and not activated on all tenants.

Please also take not that there are three images shown in the video, but four times "Add a caption" (when the page is in edit mode). There is a fourth ("invisible") image added to the page that is crucial for keeping the text from floating around the images. Does your page also contain such additional "Add a caption" element? This positioning feature has been [introduced](https://github.com/WikiTransformationProject/wikitraccs-releases/releases/tag/v0.2.0) with the March 17th release - has the page been migrated with an older version of WikiTraccs?

For future reference, here's the problems with tables I was talking about earlier: https://github.com/SharePoint/sp-dev-docs/issues/9160. Still needs to be found out if that transformation is also happening here, screwing with image positioning, or if something else is up.

**Comment by James6M** [comment]
Later in the same day that this thread was started, the behavior of the editor changed, and the problem went away (as communicated to @heinrich-ulbricht  via email).  However, the problem re-appeared several days later, and appears to be the dominant behavior.
I've now sent the requested info to contact @…

**Reply by heinrich-ulbricht** [reply]
Thank you very much for the information sent via e-mail. The screenshot confirms that the page has been converted to the new editor version.

In the meantime, I created a minimal [repro PowerShell script](https://github.com/WikiTransformationProject/library/tree/main/issue-related/ckeditor/inline-images) for folks to confirm whether the issue is present or absent, opened a support case with Microsoft (#2310071420000151) and put all known information into a blog post highlighting the behavior: [Broken inline image positioning in SharePoint](https://www.wikitransformationproject.com/blog/2023/10/07/broken-inline-image-positioning-in-sharepoint).

You also mentioned that it would be better (from an internal comms perspective) if the pages wouldn't show adjacent images in the first place, even in view mode. As soon as the new page format is GA the page transformation logic of WikiTraccs will be adjusted to the new logic and an option will be added to control this behavior.

Unfortunately there is nothing more that I can do at this point but to hope for Microsoft to fix and/or enhance the text editor web part.
